MobileCenter/MobileCenter/Internals/Channel/MSLogManagerDefaultPrivate.h

```diff
@@ -26,6 +26,33 @@ static short *const kMSStorageMaxCapacity = 300;
  */
 - (instancetype)initWithSender:(MSHttpSender *)sender storage:(id<MSStorage>)storage;
 
+/**
+ * Identifier for the background task for flushing our queue in case the app is backgrounded. We're not using
+ * UIBackgroundTaskIdentifier as it is not available on macOS and it's a typedef for NSUInteger anyway.
+ */
+@property(nonatomic) NSUInteger backgroundTaskIdentifier;
+
+/**
+ * Lock token for background task synchronization.
+ */
+@property(nonatomic, nonnull) NSObject *backgroundTaskLockToken;
+
+/**
+ * A property to hold the observer to get notified once the app enters the foreground again.
+ */
+@property(nonatomic, weak, nullable) id appWillEnterForegroundObserver;
+
+/**
+ * A property to hold the observer to get notified once the app goes into the background. It will trigger a call to
+ * flush the queue and send events to the backend.
+ */
+@property(nonatomic, weak, nullable) id appDidEnterBackgroundObserver;
+
+/**
+ * Keep track of the number of channels that are done flushing.
+ */
+@property(nonatomic) ushort flushedChannelsCount;
```

Comment:
I found the choice of `ushort` here but I guess it's fine as we're not using THAT many channels. What do you think?

Reply:
Yes that's 65 535 max channels, I felt it's good enough :slightly_smiling_face:.